File: src/main/java/examples/file/File.java
package examples.file;

import java.io.FileWriter;
import java.io.IOException;
import java.io.Writer;

import cspom.CSPOM;
import cspom.CSPParseException;

public final class File {
	private static final String FILENAME = "crossword-m1-debug-05-01.xml";

	private File() {
	}

	public static void main(final String[] args) throws IOException,
			CSPParseException {
		final CSPOM cspom;
		try {
			cspom = CSPOM.load(File.class.getResource(FILENAME));

		} catch (CSPParseException e) {
            System.err
                    .println(e.getMessage() + " at line " + e.lineNumber());
			throw e;
		}
		System.out.println(cspom);
		final Writer writer = new FileWriter("fapp.gml");
		try {
			writer.append(cspom.toGML());
		} finally {
			writer.close();
		}

	}
}

File: src/main/scala/scalaexamples/file/File.scala
package scalaexamples.file;

import cspom.{ CSPParseException, CSPOM }
import java.io.FileWriter

final object File {
  val FILENAME = "crossword-m1-debug-05-01.xml";

  def main(args: Array[String]) {
    val cspom = try CSPOM.load(File.getClass.getResource(FILENAME)) catch {
      case e: CSPParseException =>
        println(e.getMessage() + " at line " + e.lineNumber);
        throw e;
    }
    println(cspom);

    val writer = new FileWriter(FILENAME + ".gml");
    try {
      writer.append(cspom._1.toGML);
    } finally {
      writer.close();
    }

  }
}